docs/application/native/guides/ui/efl/wearable/component-gengrid.md

```diff
@@ -0,0 +1,224 @@
+# Gengrid
+
+The gengrid UI component is based on the same idea as [genlist](component-genlist.md). It aims to display objects on a grid layout and render only the visible ones. For more information, see the [Gengrid](../../../../api/wearable/latest/group__Elm__Gengrid__Group.html) API.
+
+To save memory and speed up processing when many items exist, the gengrid uses the concept of "realization" when managing items. It means that a gengrid item creates its text and content (realizes) when the user scrolls the grid and the item shows up on the screen, and frees them (unrealizes) when the item is scrolled out of the screen. To enable the item realization, you must create and fill an `Elm_Gengrid_Item_Class` structure (gengrid item class) that informs the gengrid component which callbacks to call when an item is created or deleted. When the item is created, the text and content are retrieved by calling the `text_get` and `content_get` functions defined in the gengrid item class.
+
+## Basic Usage
+
+To use a gengrid component in your application:
+
+1. Add a gengrid with the `elm_gengrid_add()` function:
+
+   ```
+   Evas_Object *gengrid;
+
+   gengrid = elm_gengrid_add(parent);
+   ```
+
+2. Define the gengrid item class:
+
+   1. Create a gengrid item class with the `elm_gengrid_item_class_new()` function, set a style to the item class, and register the callback functions.
+
+      The gengrid item class must be freed manually with `elm_gengrid_item_class_free()`  after all items are appended:
+
+      ```
+      Elm_Gengrid_Item_Class *itc = elm_gengrid_item_class_new();
+
+      itc->item_style = "default";
+      itc->func.text_get = _item_label_get;
+      itc->func.content_get = _item_content_get;
+      itc->func.del = _item_del;
+      ```
+
+   2. Define the `text_get` function.
+
+      When a gengrid item becomes realized, the `text_get` function is called repeatedly for all text parts in that item. After the text is set to the part, it is freed automatically and thus manual intervention is not needed. 
+
+      ```
+      static char*
+      _item_label_get(void *data, Evas_Object *obj, const char *part)
+      {
+          if (!strcmp(part, "elm.text"))
+              return strdup("text");
+          else
+              return NULL;
+      }
+      ```
+
+   3. Define the `content_get` function.
+
+      The `content_get` function is called repeatedly for all swallow parts in the item. It must return a valid object handle to be set, or `NULL` when no content is desired. The object is deleted by the gengrid on its deletion or when the item is unrealized.
```

Comment:
```suggestion 
      The `content_get` function is called repeatedly for all swallow parts in the item. It must return a valid object handle to be set, or `NULL` when no content is desired. The object is deleted by the gengrid on its deletion or when the item is unrealized:

Reply:
I've already fixed this, so it is outdated.